extensions/sql-migration/src/api/utils.ts

```diff
@@ -92,34 +95,69 @@ export function convertTimeDifferenceToDuration(startTime: Date, endTime: Date):
-			});
-	} else if (statusFilter === AdsMigrationStatus.SUCCEEDED) {
-		filteredMigration = databaseMigrations.filter(
-			value => value.properties?.migrationStatus === MigrationStatus.Succeeded);
-	} else if (statusFilter === AdsMigrationStatus.FAILED) {
-		filteredMigration = databaseMigrations.filter(
-			value =>
-				value.properties?.migrationStatus === MigrationStatus.Failed ||
-				value.properties?.provisioningState === ProvisioningState.Failed);
-	} else if (statusFilter === AdsMigrationStatus.COMPLETING) {
-		filteredMigration = databaseMigrations.filter(
-			value => value.properties?.migrationStatus === MigrationStatus.Completing);
-	}
-	if (databaseNameFilter) {
-		const filter = databaseNameFilter.toLowerCase();
+export function getMigrationTime(migrationTime: string): string {
+	return migrationTime
+		? new Date(migrationTime).toLocaleString()
+		: DefaultSettingValue;
+}
+
+export function getMigrationDuration(startDate: string, endDate: string): string {
+	if (startDate) {
+		if (endDate) {
+			return convertTimeDifferenceToDuration(
+				new Date(startDate),
+				new Date(endDate));
+		} else {
+			return convertTimeDifferenceToDuration(
+				new Date(startDate),
+				new Date());
+		}
+	}
+
+	return DefaultSettingValue;
+}
+
+export function filterMigrations(databaseMigrations: azure.DatabaseMigration[], statusFilter: string, columnTextFilter?: string): azure.DatabaseMigration[] {
+	let filteredMigration: azure.DatabaseMigration[] = databaseMigrations || [];
+	if (columnTextFilter) {
+		const filter = columnTextFilter.toLowerCase();
 		filteredMigration = filteredMigration.filter(
-			migration => migration.name?.toLowerCase().includes(filter));
+			migration => migration.properties.sourceServerName?.toLowerCase().includes(filter)
+				|| migration.properties.sourceDatabaseName?.toLowerCase().includes(filter)
+				|| getMigrationStatus(migration)?.toLowerCase().includes(filter)
+				|| getMigrationMode(migration)?.toLowerCase().includes(filter)
+				|| getMigrationTargetType(migration)?.toLowerCase().includes(filter)
+				|| azure.getResourceName(migration.properties.scope)?.toLowerCase().includes(filter)
+				|| azure.getResourceName(migration.id)?.toLowerCase().includes(filter)
+				|| getMigrationDuration(
+					migration.properties.startedOn,
+					migration.properties.endedOn)?.toLowerCase().includes(filter)
+				|| getMigrationTime(migration.properties.startedOn)?.toLowerCase().includes(filter)
+				|| getMigrationTime(migration.properties.endedOn)?.toLowerCase().includes(filter)
+				|| getMigrationMode(migration)?.toLowerCase().includes(filter));
+	}
+
+	switch (statusFilter) {
+		case AdsMigrationStatus.ALL:
+			return filteredMigration;
+		case AdsMigrationStatus.ONGOING:
+			return filteredMigration.filter(
+				value => {
+					const status = getMigrationStatus(value);
+					return status === MigrationStatus.InProgress
+						|| status === MigrationStatus.Completing
```

Comment:
Do we need 'Completing' here if it already has it already has its own filter type below on line 158?

Reply:
good catch